Add SimulatorOptions.set_backend method (#876)
Qiskit/qiskit-ibm-runtime: docs/how_to/noisy_simulators.rst

    options.optimization_level = 0
    options.resilience_level = 0

> :meth:`Options.simulator.set_backend` is the syntax sugar of the above set options.

Reviewer: ```suggestion
`set_backend <https://qiskit.org/ecosystem/ibm-runtime/stubs/qiskit_ibm_runtime.options.SimulatorOptions.html>`_ is the syntactic sugar for setting options.
```
